bundles/GraphicServiceBilling/Service/GraphicServiceBillingService.php

```diff
@@ -0,0 +1,293 @@
+<?php
+
+/*
+ * This file is part of aakb/jira_economics.
+ *
+ * (c) 2019 ITK Development
+ *
+ * This source file is subject to the MIT license.
+ */
+
+namespace GraphicServiceBilling\Service;
+
+use Billing\Service\BillingService;
+use PhpOffice\PhpSpreadsheet\Spreadsheet;
+
+class GraphicServiceBillingService
+{
+    private $boundProjectId;
+    private $billingService;
+    private $boundReceiverAccount;
+    private $boundReceiverPSP;
+    private $boundMaterialId;
+    private $boundWorklogPricePerHour;
+
+    /**
+     * GraphicServiceBillingService constructor.
+     *
+     * @param $boundProjectId
+     * @param \Billing\Service\BillingService $billingService
+     * @param $boundReceiverAccount
+     * @param $boundReceiverPSP
+     * @param $boundMaterialId
+     * @param $boundWorklogPricePerHour
+     */
+    public function __construct(
+        $boundProjectId,
+        BillingService $billingService,
+        $boundReceiverAccount,
+        $boundReceiverPSP,
+        $boundMaterialId,
+        $boundWorklogPricePerHour
+    ) {
+        $this->boundProjectId = $boundProjectId;
+        $this->billingService = $billingService;
+        $this->boundReceiverAccount = $boundReceiverAccount;
+        $this->boundReceiverPSP = $boundReceiverPSP;
+        $this->boundMaterialId = $boundMaterialId;
+        $this->boundWorklogPricePerHour = $boundWorklogPricePerHour;
+    }
+
+    /**
+     * Create export data for the given interval.
+     *
+     * @param \DateTime $from      start of interval
+     * @param \DateTime $to        end of interval
+     * @param bool      $marketing marketing account or not
+     *
+     * @return array
+     *
+     * @throws \Exception
+     */
+    public function createExportData(\DateTime $from, \DateTime $to, bool $marketing)
+    {
+        // Get all tasks in the interval from the project that have not been
+        // billed and that have the status "Order completed and sent".
+        $tasks = $this->getAllNonBilledFinishedTasks($this->boundProjectId, $from, $to, $marketing);
+
+        // Get debtor custom field.
+        $debtorCustomFieldId = $this->billingService->getCustomFieldId('Debitor');
+
+        $entries = [];
+
+        foreach ($tasks as $task) {
+            // Strip file link and \\ from description.
+            $description = $task->fields->description;
+            $description = preg_replace('/\\\\ \[Åbn filer i OwnCloud.*]\\ /i', '', $description);
```

Comment:
Are we sure that it's always “[Åbn filer i OwnCloud” …?

Reply:
The data is added as one big description atm. with the element always present. A better approach would be to split the order data into separate fields in Jira that could be extracted here.